crates/nu-command/src/system/run_external.rs

```diff
@@ -66,679 +188,467 @@ impl Command for External {
-                            // TODO: clean this up with an if-let chain once those are stable
-                            if let Ok(path) =
-                                nu_engine::env::path_str(engine_state, stack, self.name.span)
-                            {
-                                if let Some(cwd) = self.env_vars.get("PWD") {
-                                    // append cwd to PATH so `which-rs` looks in the cwd too.
-                                    // this approximates what cmd.exe does.
-                                    let path_with_cwd = format!("{};{}", cwd, path);
-                                    if let Ok(which_path) =
-                                        which::which_in(&self.name.item, Some(path_with_cwd), cwd)
-                                    {
-                                        if let Some(file_name) = which_path.file_name() {
-                                            if !file_name
-                                                .to_string_lossy()
-                                                .eq_ignore_case(command_name)
-                                            {
-                                                // which-rs found an executable file with a slightly different name
-                                                // than the one the user tried. Let's try running it
-                                                let mut new_command = self.clone();
-                                                new_command.name = Spanned {
-                                                    item: file_name.to_string_lossy().to_string(),
-                                                    span: self.name.span,
-                                                };
-                                                let (cmd, new_reader) = new_command
-                                                    .create_process(stdin, true, head)?;
-                                                reader = new_reader;
-                                                child = ForegroundChild::spawn(cmd);
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                    }
-
-                    child
-                }
-            };
+    Ok(result)
+}
 
-            (child, reader, input)
-        };
+/// Transforms `--option="value"` into `--option=value`. `value` can be quoted
+/// with double or single quotes. The original string should have an `=` sign,
+/// otherwise this function returns the original string. Does not resolve escape
+/// sequences within `value`. Only removes the first matching pair of quotes.
+fn remove_inner_quotes(arg: impl Into<String>) -> String {
+    let arg = arg.into();
+    let re = Regex::new(r#"^(?<option>.*?)=['"](?<value>.*)['"]$"#).expect("valid regex");
```

Comment:
The concept of quote-removal comes from the [Bash manual](https://www.man7.org/linux/man-pages/man1/bash.1.html#EXPANSION).

> After these expansions are performed, quote characters present in the original word are removed unless they have been quoted themselves (quote removal).

Bash will remove quotes very aggressively, regardless of whether the string starts with `--` or has a `=`. For example, `foo""""` is transformed into `foo`.

Do we want to follow Bash's behavior? It's not very intuitive, but it might be the least surprising because people are used to it.